Mage.Sets/src/mage/cards/g/GruffTriplets.java

```diff
@@ -0,0 +1,73 @@
+package mage.cards.g;
+
+import mage.MageInt;
+import mage.abilities.common.DiesSourceTriggeredAbility;
+import mage.abilities.common.EntersBattlefieldTriggeredAbility;
+import mage.abilities.condition.common.SourceMatchesFilterCondition;
+import mage.abilities.decorator.ConditionalInterveningIfTriggeredAbility;
+import mage.abilities.dynamicvalue.common.SourcePermanentPowerCount;
+import mage.abilities.effects.CreateTokenCopySourceEffect;
+import mage.abilities.effects.common.counter.AddCountersAllEffect;
+import mage.abilities.keyword.TrampleAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.SubType;
+import mage.counters.CounterType;
+import mage.filter.FilterPermanent;
+import mage.filter.common.FilterControlledPermanent;
+import mage.filter.predicate.mageobject.NameLikeSourcePredicate;
+import mage.filter.predicate.permanent.TokenPredicate;
+
+import java.util.UUID;
+
+/**
+ * @author Susucr
+ */
+public final class GruffTriplets extends CardImpl {
+
+    private static final FilterPermanent filterNonToken = new FilterPermanent("non-token permanent");
+    private static final FilterControlledPermanent filterNamedThis = new FilterControlledPermanent("creature you control named {this}");
+
+    static {
+        filterNonToken.add(TokenPredicate.FALSE);
+        filterNamedThis.add(NameLikeSourcePredicate.instance);
```

Comment:
this shouldn't actually check the name of the source, it's always "Gruff Triplets" even if the name of the permanent is different

Reply:
Oh I did not know that was how those effects worked. Thanks for noticing.